lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -1313,6 +1319,29 @@ export class NormalComponent<
     const accumulatedRotation =
       (decomposedTransform.rotation.angle * 180) / Math.PI
 
+    const pcbRotation =
+      pcb_component?.rotation !== undefined
+        ? pcb_component.rotation
+        : accumulatedRotation
+
+    // NEW: inherit rotation from parent group if defined
+    let groupRotation = 0
+    if (
+      this.parent?.props?.pcbRotation &&
+      typeof this.parent.props.pcbRotation === "number"
+    ) {
+      groupRotation = this.parent.props.pcbRotation
```

Comment:
This isnt right. There is a method to get the transform if the parent. This will fail with nesting

Reply:
If you need the rotation from the transform, there is a decompose method in transformation – matrix